src/map_extras.cpp

```diff
@@ -76,6 +76,165 @@ void mx_null( map &, const tripoint & )
+                    if( !one_in( 5 ) ) {
+                        m.make_rubble( tripoint( x,  y, abs_sub.z ), f_wreckage, true );
+                        if( m.ter( tripoint( x, y, abs_sub.z ) )->has_flag( TFLAG_DIGGABLE ) ) {
+                            m.ter_set( tripoint( x, y, abs_sub.z ), t_dirtmound );
+                        }
+                    } else if( m.is_bashable( x, y ) ) {
+                        m.destroy( tripoint( x,  y, abs_sub.z ), true );
+                        if( m.ter( tripoint( x, y, abs_sub.z ) )->has_flag( TFLAG_DIGGABLE ) ) {
+                            m.ter_set( tripoint( x, y, abs_sub.z ), t_dirtmound );
+                        }
+                    }
+
+                } else if( one_in( 4 + ( abs( x - cx ) + ( abs( y -
+                                         cy ) ) ) ) ) { // 1 in 10 chance of being wreckage anyway
+                    m.make_rubble( tripoint( x,  y, abs_sub.z ), f_wreckage, true );
+                    if( !one_in( 3 ) ) {
+                        if( m.ter( tripoint( x, y, abs_sub.z ) )->has_flag( TFLAG_DIGGABLE ) ) {
+                            m.ter_set( tripoint( x, y, abs_sub.z ), t_dirtmound );
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    int dir1 = rng( 0, 359 );
+
+    auto crashed_cars = vgroup_id( "city_wrecks" )->pick();
+
+    // Create the vehicle so we can rotate it and calculate its bounding box, but don't place it on the map.
+    auto veh = std::unique_ptr<vehicle>( new vehicle( city_wrecks, rng( 1, 33 ), 1 ) );
+
+    veh->turn( dir1 );
+
+    bounding_box bbox = veh->get_bounding_box();     // Get the bounding box, centered on mount(0,0)
+    int x_length = std::abs( bbox.p2.x -
+                             bbox.p1.x );  // Move the wreckage forward/backward half it's length so
+    int y_length = std::abs( bbox.p2.y -   // that it spawns more over the center of the debris area
+                             bbox.p1.y );
+
+    int x_offset = veh->dir_vec().x * ( x_length / 2 );   // cont.
+    int y_offset = veh->dir_vec().y * ( y_length / 2 );
+
+    int x_min = abs( bbox.p1.x ) + 0;
+    int y_min = abs( bbox.p1.y ) + 0;
+
+    int x_max = ( SEEX * 2 ) - ( bbox.p2.x + 1 );
+    int y_max = ( SEEY * 2 ) - ( bbox.p2.y + 1 );
+
+    int x1 = clamp( cx + x_offset, x_min,
+                    x_max ); // Clamp x1 & y1 such that no parts of the vehicle extend
+    int y1 = clamp( cy + y_offset, y_min, y_max ); // over the border of the submap.
+
+    vehicle *wreckage = m.add_vehicle( city_wrecks, tripoint( x1, y1, abs_sub.z ), dir1, rng( 1, 33 ),
+                                       1 );
+
+    const auto controls_at = []( vehicle * wreckage, const tripoint & pos ) {
+        return !wreckage->get_parts_at( pos, "CONTROLS", part_status_flag::any ).empty() ||
+               !wreckage->get_parts_at( pos, "CTRL_DASHBOARD", part_status_flag::any ).empty():
+    };
+
+    if( wreckage != nullptr ) {
```

Comment:
I suggest an early return here (`if(!wreckage) return;`). That way, you save a whole level of indentation for the below code.